kevm-pyk/src/kevm_pyk/utils.py

```diff
@@ -102,22 +185,53 @@ def run_prover(
     counterexample_info: bool = False,
     always_check_subsumption: bool = False,
     fast_check_subsumption: bool = False,
+    max_frontier_parallel: int = 1,
 ) -> bool:
     prover: APRProver | ImpliesProver
     try:
         if type(proof) is APRProof:
-            prover = APRProver(
-                kcfg_explore,
-                execute_depth=max_depth,
-                terminal_rules=terminal_rules,
-                cut_point_rules=cut_point_rules,
-                counterexample_info=counterexample_info,
-                always_check_subsumption=always_check_subsumption,
-                fast_check_subsumption=fast_check_subsumption,
-            )
-            prover.advance_proof(proof, max_iterations=max_iterations, fail_fast=fail_fast)
+
+            strategy: APRProofStrategy
```

Comment:
Consider extracting a a function for instantiating the strategy.

Reply:
Added `select_apr_strategy(...) -> APRProofStrategy` function.